actors/verifreg/tests/verifreg_actor_test.rs

```diff
@@ -544,60 +547,128 @@ mod claims {
-
-        // get claims
-        //successfully
-        let succ_gc = h.get_claims(&mut rt, PROVIDER1, vec![1, 2, 3]).unwrap();
-        assert_eq!(3, succ_gc.batch_info.success_count);
-        assert_eq!(claim2, succ_gc.claims[1]);
-
-        // bad provider
-        let fail_gc = h.get_claims(&mut rt, PROVIDER1 + 42, vec![1, 2, 3]).unwrap();
-        assert_eq!(0, fail_gc.batch_info.success_count);
-
-        // mixed bag
-        let mix_gc = h.get_claims(&mut rt, PROVIDER1, vec![1, 4, 5]).unwrap();
-        assert_eq!(1, mix_gc.batch_info.success_count);
-        assert_eq!(claim1, succ_gc.claims[0]);
+            assert_eq!(ret.codes(), vec![ExitCode::OK, ExitCode::OK]);
+            assert_alloc_claimed(&rt, CLIENT1, PROVIDER1, 1, &alloc1, 0, sector);
+            assert_alloc_claimed(&rt, CLIENT2, PROVIDER1, 2, &alloc2, 0, sector);
+        }
+        {
+            // Can't find claim for wrong client
+            rt.replace_state(&prior_state);
+            let mut reqs = vec![
+                make_claim_req(1, &alloc1, sector, expiry),
+                make_claim_req(2, &alloc2, sector, expiry),
+            ];
+            reqs[1].client = CLIENT1;
+            let ret = h.claim_allocations(&mut rt, PROVIDER1, reqs, size).unwrap();
+            assert_eq!(ret.codes(), vec![ExitCode::OK, ExitCode::USR_NOT_FOUND]);
+            assert_alloc_claimed(&rt, CLIENT1, PROVIDER1, 1, &alloc1, 0, sector);
+            assert_allocation(&rt, CLIENT2, 2, &alloc2);
+        }
+        {
+            // Can't claim for other provider
+            rt.replace_state(&prior_state);
+            let reqs = vec![
+                make_claim_req(2, &alloc2, sector, expiry),
+                make_claim_req(3, &alloc3, sector, expiry), // Different provider
+            ];
+            let ret = h.claim_allocations(&mut rt, PROVIDER1, reqs, size).unwrap();
+            assert_eq!(ret.codes(), vec![ExitCode::OK, ExitCode::USR_FORBIDDEN]);
+            assert_alloc_claimed(&rt, CLIENT1, PROVIDER1, 2, &alloc2, 0, sector);
+            assert_allocation(&rt, CLIENT1, 3, &alloc3);
+        }
+        {
+            // Mismatched data / size
+            rt.replace_state(&prior_state);
+            let mut reqs = vec![
+                make_claim_req(1, &alloc1, sector, expiry),
+                make_claim_req(2, &alloc2, sector, expiry),
+            ];
+            reqs[0].data =
+                Cid::from_str("bafyreibjo4xmgaevkgud7mbifn3dzp4v4lyaui4yvqp3f2bqwtxcjrdqg4")
+                    .unwrap();
+            reqs[1].size = PaddedPieceSize(size + 1);
+            let ret = h.claim_allocations(&mut rt, PROVIDER1, reqs, 0).unwrap();
+            assert_eq!(ret.codes(), vec![ExitCode::USR_FORBIDDEN, ExitCode::USR_FORBIDDEN]);
+        }
+        {
+            // Expired allocation
+            rt.replace_state(&prior_state);
+            let reqs = vec![make_claim_req(1, &alloc1, sector, expiry)];
+            rt.set_epoch(alloc1.expiration);
```

Comment:
Hmm I think this test will fail.  I [recently changed expiration](https://github.com/filecoin-project/builtin-actors/pull/715/files#diff-d3ee10a93cdca54a99f37a1b78a40c6b34894b097776d97030cf360eaaf165ecR1107) in response to test failures to be the last allowed epoch you can claim the allocation.  I did this so that we maintain the same behavior of ProveCommit succeeding on deal start time.  Now I'm thinking the right thing to do is instead use the meaning of expiration you expect here but change the market actor behavior to make allocation expiration be deal start + 1.

Reply:
Your interpretation matches the comment for Allocation.Expiration that is in the FIP. Even though the name is slighly off, I'll change this test to use expiration+1. Matching the deal semantics also makes sense.

Reply:
I'll leave it up to merge in #723 and tweak the FIP or change this to expiration + 1 and close down #723.

Reply:
#723 matches deal semantics while using the interpretation of the name "expiration" you are using here which is more natural imo

Reply:
Thanks, you accidentally convinced me that your original, without all the +1s, is simpler and better :-)